Source/DafnyTestGeneration/Utils.cs

```diff
@@ -55,57 +55,55 @@ public static class Utils {
     /// <summary>
     /// Parse a string read (from a certain file) to a Dafny Program
     /// </summary>
-    public static Program/*?*/ Parse(string source, string fileName = "") {
+    public static Program/*?*/ Parse(DafnyOptions options, string source, string fileName = "") {
       ModuleDecl module = new LiteralModuleDecl(new DefaultModuleDefinition(), null);
-      var builtIns = new BuiltIns();
-      var reporter = new ConsoleErrorReporter();
+      var builtIns = new BuiltIns(options);
+      var reporter = new ConsoleErrorReporter(options);
       var success = Parser.Parse(source, fileName, fileName, null, module, builtIns,
         new Errors(reporter)) == 0 && Microsoft.Dafny.Main.ParseIncludesDepthFirstNotCompiledFirst(module, builtIns,
         new HashSet<string>(), new Errors(reporter)) == null;
       Program/*?*/ program = null;
       if (success) {
-        program = new Program(fileName, module, builtIns, reporter, DafnyOptions.Create());
+        program = new Program(fileName, module, builtIns, reporter);
       }
       if (program == null) {
         return null;
       }
       // Substitute function methods with function-by-methods
-      new AddByMethodRewriter(new ConsoleErrorReporter()).PreResolve(program);
-      program.Reporter = new ErrorReporterSink();
+      new AddByMethodRewriter(new ConsoleErrorReporter(options)).PreResolve(program);
+      program.Reporter = new ErrorReporterSink(options);
       new Resolver(program).ResolveProgram(program);
       return program;
     }
 
     /// <summary>
     /// Deep clone a Boogie program.
     /// </summary>
-    public static Microsoft.Boogie.Program
-      DeepCloneProgram(Microsoft.Boogie.Program program) {
-      var textRepresentation = GetStringRepresentation(program);
+    public static Microsoft.Boogie.Program DeepCloneProgram(DafnyOptions options, Microsoft.Boogie.Program program) {
+      var textRepresentation = GetStringRepresentation(options, program);
       Microsoft.Boogie.Parser.Parse(textRepresentation, "", out var copy);
       return copy;
     }
 
     /// <summary>
     /// Deep clone and re-resolve a Boogie program.
     /// </summary>
-    public static Microsoft.Boogie.Program
-      DeepCloneProgramAndReresolve(Microsoft.Boogie.Program program, DafnyOptions options) {
-      program = DeepCloneProgram(program);
+    public static Microsoft.Boogie.Program DeepCloneResolvedProgram(Microsoft.Boogie.Program program, DafnyOptions options) {
```

Comment:
Now I see the answer to my previous question! I'm not sure that this name change is an improvement in clarity.

Reply:
Fair enough!